plugins/modules/net_tools/ldap/ldap_search.py

```diff
@@ -168,10 +168,14 @@ def perform_search(self):
                 attrlist=self.attrlist,
                 attrsonly=self.attrsonly
             )
-            if self.schema:
-                return [dict(dn=result[0], attrs=list(result[1].keys())) for result in results]
-            else:
-                return [_extract_entry(result[0], result[1]) for result in results]
+            ldap_entries = []
+            for result in results:
+                if isinstance(result[1], dict):
```

Comment:
Test return from ldap query to not parse referral as an ldap entry. See issue #963.